app/models/report_filter.rb

```diff
@@ -5,38 +5,67 @@ class ReportFilter
     resolved
     account_id
     target_account_id
-    by_target_domain
+    search_type
+    search_term
     target_origin
   ).freeze
 
+  IGNORED_PARAMS = %i(resolved search_type search_term).freeze
+
   attr_reader :params
 
   def initialize(params)
     @params = params
   end
 
   def results
-    scope = Report.unresolved
+    scope = if params[:resolved] == '-1'
+              Report.unscoped
+            elsif params[:resolved] == '1'
+              Report.resolved
+            else
+              Report.unresolved
+            end
```

Comment:
Should probably change this to `params[:status]`, and then add a redirect that replaces `resolved=1` with `status=resolved`, since the `-1` is absolutely a hack here.

Reply:
Have done this.